packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -355,3 +375,23 @@ abstract class _ExtensionsApiHandler {
     return ServerApi._encodeResponse(activationState.name, api: api);
   }
 }
+
+abstract class _DeeplinkApiHandler {
+  static Future<shelf.Response> handleAndroidBuildVariants(
+    ServerApi api,
+    Map<String, String> queryParams,
+    DeeplinkManager deeplinkManager,
+  ) async {
+    final missingRequiredParams = ServerApi._checkRequiredParameters(
+      [DeeplinkApi.deeplinkRootPathPropertyName],
+      queryParams: queryParams,
+      api: api,
+      requestName: DeeplinkApi.androidBuildVariants,
+    );
+    if (missingRequiredParams != null) return missingRequiredParams;
+
+    final rootPath = queryParams[DeeplinkApi.deeplinkRootPathPropertyName]!;
+
+    return deeplinkManager.getBuildVariants(rootPath: rootPath, api: api);
```

Comment:
instead of having `getBuildVariants` return `api.getCompleted`, let's have `getBuildVariants` return the output json, either encoded as a string or as a raw map, and then return `api.getCompleted` here. This will remove the need for `devtools_deeplink.dart` to have to depend on the server_api and will be consistent with the other api endpoints.